### PR TITLE
Add full json of a Ktor Feature to be stored as blob in Marketplace

### PR DESCRIPTION
**Pull request by Ololoshechkin**

### CODE REVIEW

File: intellij-plugin-structure/structure-ktor/src/main/kotlin/com/jetbrains/plugin/structure/ktor/KtorFeaturePluginManager.kt [hunk 1]
```diff
@@ -104,8 +104,8 @@ class KtorFeaturePluginManager private constructor(private val extractDirectory:
               usage = doc.usage!!,
               options = doc.options!!
             )
-          }
-
+          },
+          fullDescriptorJson = Json.stringify(KtorFeatureDescriptor.serializer(), descriptor)
```
Comment (AlexanderPrendota) [inline intellij-plugin-structure/structure-ktor/src/main/kotlin/com/jetbrains/plugin/structure/ktor/KtorFeaturePluginManager.kt:108]: Do you need? JsonConfiguration.Stable.copy(isLenient = true, ignoreUnknownKeys = true)

File: intellij-plugin-structure/tests/build.gradle [hunk 2]
```diff
@@ -1,3 +1,7 @@
+plugins {
+  id "org.jetbrains.kotlin.plugin.serialization"
```
Comment (AlexanderPrendota) [inline intellij-plugin-structure/tests/build.gradle:2]: we use jackson in test(

File: intellij-plugin-structure/tests/build.gradle [hunk 3]
```diff
@@ -12,6 +16,7 @@ dependencies {
   testImplementation 'junit:junit:4.11'
   testImplementation "com.fasterxml.jackson.module:jackson-module-kotlin:2.11.0"
   testImplementation "com.google.jimfs:jimfs:1.1"
-
+  testImplementation "com.google.jimfs:jimfs:1.1"
```
Comment (AlexanderPrendota) [inline intellij-plugin-structure/tests/build.gradle:19]: why do u need?

File: intellij-plugin-structure/structure-ktor/src/main/kotlin/com/jetbrains/plugin/structure/ktor/KtorFeature.kt [hunk 4]
```diff
@@ -17,7 +17,8 @@ data class KtorFeature(
   override val pluginVersion: String? = null,
   override val pluginId: String? = null,
   val requiredFeatures: List<String> = emptyList(), // Feature IDs.
-  val documentation: KtorFeatureDocumentation? = null
+  val documentation: KtorFeatureDocumentation? = null,
+  val fullDescriptorJson: String? = null
```
Comment (AlexanderPrendota) [inline intellij-plugin-structure/structure-ktor/src/main/kotlin/com/jetbrains/plugin/structure/ktor/KtorFeature.kt:21]: fullDescriptorJson should be not null and empty.

**Review by AlexanderPrendota — CHANGES_REQUESTED** [review]

File: intellij-plugin-structure/structure-ktor/src/main/kotlin/com/jetbrains/plugin/structure/ktor/KtorFeature.kt [hunk 5]
```diff
@@ -17,7 +17,8 @@ data class KtorFeature(
   override val pluginVersion: String? = null,
   override val pluginId: String? = null,
   val requiredFeatures: List<String> = emptyList(), // Feature IDs.
-  val documentation: KtorFeatureDocumentation? = null
+  val documentation: KtorFeatureDocumentation? = null,
+  val fullDescriptorJson: String = ""
```
Comment (AlexanderPrendota) [inline intellij-plugin-structure/structure-ktor/src/main/kotlin/com/jetbrains/plugin/structure/ktor/KtorFeature.kt:21]: missing validation

**Review by AlexanderPrendota — CHANGES_REQUESTED** [review]

File: intellij-plugin-structure/structure-ktor/src/main/kotlin/com/jetbrains/plugin/structure/ktor/KtorFeature.kt (hunk 5 above)
Comment (Ololoshechkin) [reply intellij-plugin-structure/structure-ktor/src/main/kotlin/com/jetbrains/plugin/structure/ktor/KtorFeature.kt:21]: It is being created in one single place and the field gets set there. See `KtorFeaturePluginManager:110`. So it is just a string representation of the json that is already validated by that time

**Review by AlexanderPrendota — APPROVED** [review]

**Pull request merged**